### PR TITLE
python-common: Make Drive Group filter by AND, instead of OR

### PR DESCRIPTION
**Pull request by sebastian-philipp**
python-common: Make Drive Group filter by AND, instead of OR

Right now, a drive group that specifies `size=100GB` + `rotational=True`
matches on all drives that are either `size=100GB` OR `rotational=True`.
Now we change this to AND.

Fixes: https://tracker.ceph.com/issues/43713

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/python-common/ceph/tests/test_drive_group.py [hunk 1]
```diff
@@ -70,8 +70,8 @@ def test_ceph_volume_command_2():
                           db_devices=DeviceSelection(size='200GB:350GB', rotational=False),
                           wal_devices=DeviceSelection(size='10G')
                           )
-    inventory = _mk_inventory(_mk_device(rotational=True)*2 +
-                              _mk_device(rotational=False)*2 +
+    inventory = _mk_inventory(_mk_device(rotational=True, size="300.00 GB")*2 +
+                              _mk_device(rotational=False, size="300.00 GB")*2 +
```
Comment (sebastian-philipp) [inline src/python-common/ceph/tests/test_drive_group.py]: ... this actually fixes a bug here. by default, drives are `390GB` and thus bigger than `size='200GB:350GB'`. The size attribute never matched.

**Comment by sebastian-philipp** [comment]
some pesky linting errors:
```
ceph/deployment/drive_selection/filter.py:8:5: F401 'typing.Optional' imported but unused
ceph/tests/test_disk_selector.py:5:1: F401 'ceph.deployment.drive_group.DriveGroupValidationError' imported but unused
ceph/tests/test_disk_selector.py:10:5: F401 'mock.patch' imported but unused
```

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2020-02-29_17:36:34-rados-wip-kefu-testing-2020-02-29-2106-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-03-02_05:23:09-rados-wip-kefu-testing-2020-03-01-2348-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-03-02_05:23:09-rados-wip-kefu-testing-2020-03-01-2348-distro-basic-smithi/

**Comment by tchaikov** [comment]
might want to address the lint issues..

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2020-03-07_07:27:51-rados-wip-kefu-testing-2020-03-07-1143-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-03-07_16:58:24-rados-wip-kefu-testing-2020-03-07-2053-distro-basic-smithi/

**Pull request merged**